src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleDamageParticles.kt

```diff
@@ -103,7 +107,10 @@ object ModuleDamageParticles : Module("DamageParticles", Category.RENDER) {
 
         healthMap.keys.removeIf { it !in entities || it.isDead }
 
-        particles.removeIf { now - it.startTime > ttl * 1000F }
+        val earliest = now - (ttl * 1000).toLong()
+        while (particles.isNotEmpty() && particles.first().startTime < earliest) {
+            particles.removeFirst()
+        }
```

Comment:
You found a bad usage of a datastructure and replaced it with a better one. Good optimization 👍🏻